### PR TITLE
fix-structure

### PR DESCRIPTION
**Pull request by ronbn1**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,12 @@
 import React from "react";
```
Comment (almogbhl) [inline src/App.tsx]: you dont need an app file

only index that will export the components so you can import for others who use your library

index.ts:
export { Button } from "./components/button/Button";

an so on...

in root open folder '.storybook':
config.js:
import { addDecorator } from "@storybook/react";
import themeDecorator from "./themeDecorator";

addDecorator(themeDecorator);


main.js:
const path = require("path");

module.exports = {
  stories: ["../src/**/*.stories.tsx"],
  // Add any Storybook addons you want here: https://storybook.js.org/addons/
  addons: [],
  webpackFinal: async (config) => {
    config.module.rules.push({
      test: /\.scss$/,
      use: ["style-loader", "css-loader", "sass-loader"],
      include: path.resolve(__dirname, "../")
    });

    config.module.rules.push({
      test: /\.(ts|tsx)$/,
      loader: require.resolve("babel-loader"),
      options: {
        presets: [["react-app", { flow: false, typescript: true }]]
      }
    });
    config.resolve.extensions.push(".ts", ".tsx");

    return config;
  }
};

themeDecorator.js:
import React from "react"
import { ThemeProvider } from '../src';

const ThemeDecorator = storyFn => (
  <ThemeProvider>{storyFn()}</ThemeProvider>
)

export default ThemeDecorator

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -36,12 +11,10 @@ interface Props {
...
 const Button = (props: Props) => {
    return (
-      <StyledButton color={props.color} size={props.size}>
+      <SButton color={props.color} size={props.size} icon={props.icon}>
          {props.text}
-         <span>
-            {props.icon ? <FontAwesomeIcon icon={icons[props.icon]} /> : ""}
-         </span>
-      </StyledButton>
+         {props.icon && <Sicon className="icon" icon={props.icon} />}
```
Comment (almogbhl) [inline src/components/Button/Button.tsx]: 1. try keeping the convention of upper case in the beginning of the word Sicon => SIcon
2. why we need the className if we have the style element?

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -36,12 +11,10 @@ interface Props {
 
 const Button = (props: Props) => {
```
Comment (almogbhl) [inline src/components/Button/Button.tsx]: why not Destructuring it like that
const Button = ({color, size, icon}: IButtonProps) => {

File: src/components/Button/Button.tsx [hunk 4]
```diff
@@ -1,31 +1,6 @@
...
-   faPlusCircle,
-   faArrowDown,
-   faSmile,
-   faHeart,
-   faTrash
-);
+import { SButton, Sicon } from "./style";
+import { colorType, sizesType, iconType } from "../../types/types";
 
 interface Props {
```
Comment (almogbhl) [inline src/components/Button/Button.tsx]: convention for interfaces - starting with big I and the name so:
IButtonProps could be a good one

File: src/components/Button/Button.tsx [hunk 5]
```diff
@@ -1,31 +1,6 @@
...
-   faArrowRight,
-   faArrowLeft,
-   faPlusCircle,
-   faArrowDown,
-   faSmile,
-   faHeart,
-   faTrash
-);
+import { SButton, Sicon } from "./style";
+import { colorType, sizesType, iconType } from "../../types/types";
```
Comment (almogbhl) [inline src/components/Button/Button.tsx:3]: do we use them here?

File: src/themes/online.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+   },
+   colors: {
+      secondary: "#3d5a80",
+      success: "#3bceac",
+      dark: "#293241",
+      primary: "#ee6c4d",
+      light: "#e0fbfc",
+   },
+   icons: {
+      add: plus,
```
Comment (almogbhl) [inline src/themes/online.ts]: why not name them the same?

**Review by almogbhl — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (ronbn1) [reply src/App.tsx]: done.

File: src/components/Button/Button.tsx (hunk 2 above)
Comment (ronbn1) [reply src/components/Button/Button.tsx]: 1.done
2. I need to select that element in the parent's styled component for the hover effect, if you have another suggestion I'm open for suggestions.

File: src/components/Button/Button.tsx (hunk 3 above)
Comment (ronbn1) [reply src/components/Button/Button.tsx]: done

File: src/components/Button/Button.tsx (hunk 4 above)
Comment (ronbn1) [reply src/components/Button/Button.tsx]: done

File: src/themes/online.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+   icons: {
+      plus: plus,
+      trash: trash,
+      heart: heart,
+      downArrow: downArrow,
+      rightArrow: rightArrow,
+      leftArrow: leftArrow,
+      upArrow: upArrow,
+      smile: smile,
+   },
```
Comment (almogbhl) [inline src/themes/online.ts:88]: can also be:
 icons: {
      plus,
      trash,
      heart,
      downArrow,
      rightArrow,
      leftArrow,
      upArrow,
      smile,
   },

File: src/components/Button/style.ts [hunk 8]
```diff
@@ -25,16 +27,31 @@ const StyledButton: React.FunctionComponent<Props> = styled.button`
...
+   background-size: cover;
+   display: inline-block;
+   position: relative;
+   left: 5px;
+   top: 2.5px;
+   width: 1rem;
+   height: 1rem;
+   background-color: ${(props: Props) =>
+      props.color
+         ? onlineTheme.colors[props.color]
```
Comment (almogbhl) [inline src/components/Button/style.ts]: you want to export the theme like that:
styled.span(({ theme }) => {
  return `
  background-color: ${(props: Props) =>
      props.color
         ? theme.colors[props.color]
  `;
});

**Review by almogbhl — APPROVED** [review]

**Pull request merged**